extensions/pkg/controller/dnsrecord/reconciler.go

```diff
@@ -247,34 +243,30 @@ func (r *reconciler) delete(ctx context.Context, dns *extensionsv1alpha1.DNSReco
 
 func updateCreatedCondition(status extensionsv1alpha1.Status, conditionStatus gardencorev1beta1.ConditionStatus, reason string, updateIfExisting bool) error {
 	conditions := status.GetConditions()
-	index := len(conditions)
-	for i, c := range conditions {
-		if c.Type == createdConditionType {
-			if !updateIfExisting {
-				return nil
-			}
-			index = i
-			break
-		}
-	}
-	if index == len(conditions) {
-		conditions = append(conditions, gardencorev1beta1.Condition{Type: createdConditionType})
+	c := gardencorev1beta1helper.GetCondition(conditions, extensionsv1alpha1.ConditionTypeCreated)
+	if c != nil && !updateIfExisting {
+		return nil
 	}
-	c := &conditions[index]
-	if c.Status == conditionStatus {
+	if c != nil && c.Status == conditionStatus {
 		return nil
 	}
-	c.Status = conditionStatus
-	c.LastTransitionTime = metav1.Time{Time: time.Now()}
-	c.LastUpdateTime = metav1.Time{Time: time.Now()}
-	c.Reason = reason
+	if c == nil {
+		conditions = append(conditions, gardencorev1beta1helper.InitCondition(extensionsv1alpha1.ConditionTypeCreated))
+		c = &conditions[len(conditions)-1]
+	}
+
+	builder, err := gardencorev1beta1helper.NewConditionBuilder(extensionsv1alpha1.ConditionTypeCreated)
+	if err != nil {
+		return err
+	}
+	*c, _ = builder.WithStatus(conditionStatus).WithReason(reason).WithMessage("").Build()
```

Comment:
Please pass the old condition to `WithOldCondition` so that the timestamps are correctly handled.
Can you add a message as well?

Reply:
done